src/main/kotlin/se/uulm/snowballr/backend/service/ProjectPaperService.kt

```diff
@@ -279,4 +318,40 @@ class ProjectPaperService(
 
     override suspend fun getPreviousPaper(request: Base.Id): GrpcProjectPaper =
         getAdjacentPaper(request.id, PaperNavigationDirection.PREVIOUS)
+
+    override suspend fun getNextPaperToReview(request: Base.Id): GrpcProjectPaper = withUser(userRepo) { currentUser ->
+        val projectPaperId = parseUUID(request.id, EntityType.PROJECT_PAPER)
+        val projectPaper = repo.getProjectPaperById(projectPaperId).getOrThrow()
+        val projectId = projectPaper.projectId
+
+        isAllowedToReadProject(projectMemberRepo).checkFor(currentUser, projectId)
+
+        val projectPapers = repo.getLaterProjectPapers(
+            projectId,
+            projectPaper.localPaperId,
+            projectPaper.stage,
+        )
+        val projectPapersWithReviews: List<Pair<ProjectPaper, List<Review>>> = projectPapers
+            .map { paper ->
+                val reviews = reviewTableRepo.getAllReviewsForProjectPaper(paper.id)
```

Comment:
Here we can directly filter out own reviews so we don't have to do it in `getProjectPapersWithoutOwnReview`.

```suggestion
                val reviews = reviewTableRepo.getAllReviewsForProjectPaper(paper.id)
                     .filter { it.userId != currentUser.id }
```

Reply:
I am not sure if I got it as you meant. I have extracted is as an extra method. But it does not make sense to filter it like this there:
```kotlin
val reviews = reviewTableRepo.getAllReviewsForProjectPaper(paper.id)
                     .filter { it.userId != currentUser.id }
```
We would filter out the own reviews here, but then we are not able to filter out all the project papers which are already reviewed by the current user in the next step.

Reply:
Yes, my comment was a bit confusing. We fixed it with the refactoring.